api/authenticator.py

```diff
@@ -639,7 +658,7 @@ def create_authentication_document(self):
 
         library_name = self.library_name or unicode(_("Library"))
         doc = OPDSAuthenticationDocument.fill_in(
-            base_opds_document, list(self.providers),
+            base_opds_document, list(self.providers), self._db,
```

Comment:
I don't like that I had to do this. The database is needed to look up the library to get its short name when creating urls.